source/extensions/filters/network/tcp_proxy/config.cc

```diff
@@ -22,9 +22,7 @@ Network::FilterFactoryCb ConfigFactory::createFilterFactoryFromProtoTyped(
     const envoy::config::filter::network::tcp_proxy::v2::TcpProxy& proto_config,
     Server::Configuration::FactoryContext& context) {
   ASSERT(!proto_config.stat_prefix().empty());
-  if (proto_config.has_deprecated_v1()) {
-    ASSERT(proto_config.deprecated_v1().routes_size() > 0);
-  }
+  ASSERT(!proto_config.cluster().empty());
```

Comment:
Since there has been no constraint on this field in the proto, I don't think it's acceptable to ASSERT this. Throw instead.

Reply:
Addressed in 7d8887d.